app/bundles/FormBundle/EventListener/CustomFieldSubscriber.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\FormBundle\EventListener;
+
+use Mautic\FormBundle\Entity\FieldRepository;
+use Mautic\FormBundle\Model\FormModel;
+use Mautic\LeadBundle\Event\LeadFieldEvent;
+use Mautic\LeadBundle\LeadEvents;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+final class CustomFieldSubscriber implements EventSubscriberInterface
+{
+    public function __construct(
+        private FieldRepository $fieldRepository,
+        private FormModel $formModel,
+    ) {
+    }
+
+    /**
+     * @return mixed[]
+     */
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            LeadEvents::FIELD_POST_SAVE => ['regenerateFormCache', 0],
+        ];
+    }
+
+    /**
+     * If a custom (contact) field options change then we need to regenerate HTML cache of the forms
+     * that use that field so that the new options are shown.
+     */
+    public function regenerateFormCache(LeadFieldEvent $event): void
```

Comment:
I do not see any     check for changes, then I assume it clear cache on each multiselect/select contact field.

```
$changes = $event->getChanges();
      if (empty($changes['properties'])) {
          return; // No properties changed, no need to regenerate
      }
```

Reply:
I was looking for the changes via Xdebug but it wasn't available. Do you see them?